src/auth.js

```diff
@@ -1,197 +1,161 @@
-}
-
-/**
- * Builds an auth entry with a signed invocation tree.
+ * @returns {Promise<xdr.SorobanAuthorizationEntry>} a promise for an
+ *    authorization entry that you can pass along to
+ *    {@link Operation.invokeHostFunction}
  *
- * You should first build the envelope using {@link buildAuthEnvelope}. If you
- * have a signing {@link Keypair}, you can use the more convenient
- * {@link authorizeInvocation} to do signing for you.
+ * @example
+ * import { Server, Transaction, Networks, authorizeEntry } from 'soroban-client';
  *
- * @param {xdr.HashIdPreimage} envelope   an envelope to represent the call tree
- *    being signed, probably built by {@link buildAuthEnvelope}
- * @param {Buffer|Uint8Array} signature   a signature of the hash of the
- *    envelope by the private key corresponding to `publicKey` (in other words,
- *    `signature = sign(hash(envelope))`)
- * @param {string} publicKey  the public identity that signed this envelope
+ * // Assume signPayloadCallback is a well-formed signing callback.
+ * //
+ * // It might, for example, pop up a modal from a browser extension, send the
+ * // transaction to a third-party service for signing, or just do simple
+ * // signing via Keypair like it does here:
+ * function signPayloadCallback(payload) {
+ *    const signature = signer.sign(hash(payload.toXDR());
+ *    return [ signature, signer.publicKey() ];
+ * }
  *
- * @returns {xdr.SorobanAuthorizationEntry}
+ * function multiPartyAuth(
+ *    server: Server,
+ *    // assume this involves multi-party auth
+ *    tx: Transaction,
+ * ) {
+ *    return server
+ *      .simulateTransaction(tx)
+ *      .then((simResult) => {
+ *          tx.operations[0].auth.map(entry =>
+ *            authorizeEntry(
+ *              entry,
+ *              signPayloadCallback,
+ *              currentLedger + 1000,
+ *              Networks.FUTURENET);
+ *          ));
  *
- * @throws {Error} if `verify(hash(envelope), signature, publicKey)` does not
- *    pass, meaning one of the arguments was not passed or built correctly
- * @throws {TypeError} if the envelope does not hold an
- *    {@link xdr.HashIdPreimageSorobanAuthorization} instance
+ *          return server.prepareTransaction(tx, Networks.FUTURENET, simResult);
+ *      })
+ *      .then((preppedTx) => {
+ *        preppedTx.sign(source);
+ *        return server.sendTransaction(preppedTx);
+ *      });
+ * }
  */
-export function buildAuthEntry(envelope, signature, publicKey) {
-  // ensure this identity signed this envelope correctly
+export async function authorizeEntry(
+  entry,
+  signingMethod,
+  validUntil,
+  networkPassphrase = Networks.FUTURENET
+) {
+  // no-op
   if (
-    !Keypair.fromPublicKey(publicKey).verify(hash(envelope.toXDR()), signature)
+    entry.credentials().switch() !==
+    xdr.SorobanCredentialsType.sorobanCredentialsAddress()
   ) {
-    throw new Error(`signature does not match envelope or identity`);
+    return entry;
```

Comment:
It might be more useful for this to fail if you try to sign something that can't be signed. idk tho

Reply:
That would make it such that you can't do a `auth.map((entry) => authorizeEntry(entry, complexAsyncSigningMethod, ...)` kinda thing, unfortunately. You could just add a `.filter` first, I guess, but I don't have a strong opinion either way either